lib/puppet/environments.rb

```diff
@@ -356,6 +356,15 @@ def list
       # Evict all that have expired, in the same way as `get`
       clear_all_expired
 
+      cached_envs = @cache.values.map!(&:value).map!(&:name)
```

Comment:
Did you not do `@cache.keys` because you're concerned the key may be different than the `Environment#name` (string vs symbol)?

Reply:
Missed that, updated to use `@cache.keys.map(&:to_sym)`